frontend/src/components/activePage/TableInstance/RowInstanceHeader/RowInstanceHeader.tsx

```diff
@@ -0,0 +1,219 @@
+  SortDescendingOutlined,
+} from '@ant-design/icons';
+
+const { Text } = Typography;
+export interface IRowInstanceHeaderProps {
+  viewMode: WorkspaceRole;
+  showGuiIcon?: boolean;
+  templatePrettyName: string;
+  handleSorting: (sortingType: string, sorting: number) => void;
+  handleManagerSorting: (
+    sortingType: string,
+    sorting: number,
+    sortingTemplate: string
+  ) => void;
+}
+
+const RowInstanceHeader: FC<IRowInstanceHeaderProps> = ({ ...props }) => {
+  const {
+    viewMode,
+    handleSorting,
+    handleManagerSorting,
+    templatePrettyName: tpn,
+  } = props;
+  const [prettyNameOrder, setPrettyNameOrder] = useState(0);
+  const [templatePrettyNameOrder, setTemplatePrettyNameOrder] = useState(0);
+  const [timeStampOrder, setTimeStampOrder] = useState(0);
+  const [tenantDisplayNameOrder, setTenantDisplayNameOrder] = useState(0);
+  const [tenantIdOrder, setTenantIdOrder] = useState(0);
+
+  type sortKey =
+    | 'tenantId'
+    | 'tenantDisplayName'
+    | 'templatePrettyName'
+    | 'timeStamp'
+    | 'prettyName';
+
+  const getSorting = (value: number) => {
+    return value === (0 || -1) ? 1 : -1;
+  };
+
+  const varByKey = {
+    tenantId: tenantIdOrder,
+    tenantDisplayName: tenantDisplayNameOrder,
+    templatePrettyName: templatePrettyNameOrder,
+    timeStamp: timeStampOrder,
+    prettyName: prettyNameOrder,
+  };
+
+  const selectOrder = (key: sortKey) => {
+    clearOthers(key);
+    switch (key) {
+      case 'tenantId':
+        setTenantIdOrder(getSorting(varByKey[key]));
+        break;
+      case 'tenantDisplayName':
+        setTenantDisplayNameOrder(getSorting(varByKey[key]));
+        break;
+      case 'templatePrettyName':
+        setTemplatePrettyNameOrder(getSorting(varByKey[key]));
+        break;
+      case 'timeStamp':
+        setTimeStampOrder(getSorting(varByKey[key]));
+        break;
+      case 'prettyName':
+        setPrettyNameOrder(getSorting(varByKey[key]));
+        break;
+      default:
+        break;
+    }
+    viewMode === WorkspaceRole.manager
+      ? handleManagerSorting(key, varByKey[key], tpn)
+      : handleSorting(key, varByKey[key]);
+  };
+
+  const clearOthers = (key: sortKey) => {
+    key !== 'templatePrettyName' && setTemplatePrettyNameOrder(0);
+    key !== 'timeStamp' && setTimeStampOrder(0);
+    key !== 'tenantId' && setTenantIdOrder(0);
+    key !== 'prettyName' && setPrettyNameOrder(0);
+    key !== 'tenantDisplayName' && setTenantDisplayNameOrder(0);
+  };
+
+  const getArrow = (value: number, alpha: boolean) => {
+    return (
+      <>
+        {value === 1 &&
+          (alpha ? (
+            <SortAscendingOutlined className="pl-1" />
+          ) : (
+            <CaretDownOutlined className="pl-1" />
+          ))}
+        {value === -1 &&
+          (alpha ? (
+            <SortDescendingOutlined className="pl-1" />
+          ) : (
+            <CaretUpOutlined className="pl-1" />
+          ))}
+      </>
+    );
```

Comment:
I think it can be better to make this plain, like
```suggestion
        if (value > 0) {
          if (alpha) return (<SortAscendingOutlined className="pl-1" />);
          return (<CaretDownOutlined className="pl-1" />);
        }
        if (value < 0) {
          if (alpha) return (<SortDescendingOutlinedclassName="pl-1" />);
          return (<CaretUpOutlinedclassName="pl-1" />);
        }
```
Better not to abuse of conditional jsx syntax